gobblin-utility/src/main/java/gobblin/util/concurrent/TaskSchedulerFactory.java

```diff
@@ -0,0 +1,54 @@
+package gobblin.util.concurrent;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Map;
+
+public class TaskSchedulerFactory {
```

Comment:
Can you explain what the rationale for this class is? It seems to be caching something, but I don't understand what.